client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import request from 'superagent';
+import defer from 'lodash/defer';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import {
+	LOGIN_TWOFACTOR_UPDATE_NONCE,
+	LOGIN_TWOFACTOR_PUSH_POLL_COMPLETED,
+	LOGIN_TWOFACTOR_PUSH_POLL_START,
+} from 'state/action-types';
+import {
+	getTwoFactorUserId,
+	getTwoFactorAuthNonce,
+	getTwoFactorPushToken,
+	getTwoFactorRememberMe,
+	getTwoFactorPushPollInProgress
+} from 'state/login/selectors';
+
+/***
+ * Module constants
+ */
+const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
+
+/***
+ * Checks the status of the push notification auth
+ *
+ * @param {Object}   store  Global redux store
+ * @returns {Promise}		Promise of result from the API
+ */
+const doAppPushRequest = ( store ) => {
```

Comment:
It looks like a regular action at the moment. I have a feeling that data layer expects a bit different style.